include/aws/s3/s3_client.h

```diff
@@ -632,6 +632,14 @@ struct aws_s3_meta_request_options {
      * Do not set the message's body-stream if the body is being passed by other means (see note above) */
     struct aws_http_message *message;
 
+    /**
+     * Optional.
+     * If set, the received data will be written into this file.
```

Comment:
Mention that we will overwrite the file if it already exists.

Reply:
good call, reminds me that we actually need to support Java SDK's file options. https://github.com/aws/aws-sdk-java-v2/blob/master/core/sdk-core/src/main/java/software/amazon/awssdk/core/FileTransformerConfiguration.java#L179

Added the support